src/libs/addEmojiToComposerTextInput/index.android.js

```diff
@@ -0,0 +1,22 @@
+import addEmojiToComposerTextInput from './addEmojiToComposerTextInput';
+
+/**
+ * Takes a text and adds an emoji at the place of selection.
+ * It will then update the text of the given TextInput using its `setTextAndSelection` method.
+ * `setTextAndSelection` method is usually available on TextInput refs from the composer component.
+ *
+ * @param {AddEmojiToComposerTextInputParams} params
+ * @return {AddEmojiToComposerTextInputReturnType}
+ */
+export default (params) => {
+    const {prevSelection, textInput} = params;
+    const hasRangeSelected = prevSelection.start !== prevSelection.end;
+    if (hasRangeSelected) {
+        // Android: when we have a range selected setSelection
+        // won't remove the highlight, so we manually set the cursor
+        // to a selection range of 0 (so there won't be any selection highlight).
+        textInput.setSelection(prevSelection.start, prevSelection.start);
```

Comment:
IMO, This might not work at all times. I will test it and share more details if a real issue.